fleet/fleet_test.go

```diff
@@ -12,12 +13,21 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+var logger = requestcontext.MustGetLogger(
```

Comment:
I don't understand this. I have this in mind. Every object created provides a mechanism to create some default configuration that can be used to create a instance of itself. That means we have `DefaultFooConfig() FooConfig` and `NewFoo Foo`. Here `Foo` uses e.g. a logger. This logger is a property of `FooConfig`. `FooConfig` is initialized by best effort in `DefaultFooConfig`. So by default there should be some instance of a logger be set in the default config. That way we don't need to do this here, because the logger is already in the default config and we simply need to do `newFoo := foo.NewFoo(foo.DefaultFooConfig())` in most cases.